src/shared/extract-error-string.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+import type {AmbiguousError} from "./types.js";
+
+/**
+ * Extract an error string from an ambiguous error.
+ *
+ * This takes an ambiguous error representation and attempts to turn it into
+ * a string for logging. If a string is passed, that string is returned.
+ * This function can be recursive.
+ *
+ * @param {mixed} error An object or string that represents an error.
+ * @returns {string} A string representing the error.
+ */
+export function extractErrorString(error: AmbiguousError): string {
+    if (typeof error === "string") {
+        return error;
+    }
+
+    if (error.response && error.response.error) {
+        return `${error.response.error}: ${error.stack || ""}`;
+    }
+
+    if (error.error && error !== error.error) {
+        return extractErrorString(error.error);
+    }
```

Comment:
I didn't realize errors could be recursive in nature.

Reply:
Yeah, this was an exciting discovery as I tried to get RRS to output useful error info.